Assets/Scripts/Enemy/Core/EnemyBehaviour.cs

```diff
@@ -0,0 +1,142 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyBehaviour : MonoBehaviour
+{
+    // PUBLIC DECLARATIONS
+    // Scripts
+    [Header("Master Script")] public EnemyController Controller;
+
+    [Header("Player Target")] public Transform PlayerTarget;
+
+    [HideInInspector] public Vector3 DirectionToPlayer;
+
+    // PRIVATE DECLARATIONS
+    private float _distanceToPlayer;
+
+    private bool _isAttackingMelee;
+
+    // Start is called before the first frame update
+    private void Start()
+    {
+        // Get all available scripts on the entity
+        Controller = GetComponent<EnemyController>();
+    }
+
+    // Update is called once per frame
+    private void Update()
+    {
+        FindDistanceToPlayer();
+        LookAtPlayer();
+    }
+
+    public Vector3 FindDirectionToPlayer()
+    {
+        Vector3 _dirToPlayer = (transform.position - Controller.PlayerEntity.transform.position).normalized;
+        return _dirToPlayer;
+    }
+
+    private void LookAtPlayer()
+    {
+        // Looks at the player's position
+        Vector3 _playerPos = new Vector3(Controller.PlayerEntity.transform.position.x,
+            0f,
+            Controller.PlayerEntity.transform.position.z);
+
+        this.transform.LookAt(_playerPos);
+
+        if (Controller.IsRanged == true)
+        {
+            Vector3 _playerHeadPos = new Vector3(0f,
+            Controller.PlayerTarget.position.y,
+            0f);
+
+            Controller.Ranged.ShotOrigin.transform.LookAt(_playerHeadPos);
+        }
+    }
+
+    private void FindDistanceToPlayer()
+    {
+        // Find the distance to player
+        _distanceToPlayer = Vector3.Distance(this.transform.position,
+            Controller.PlayerEntity.transform.position);
+
+        // Checks if the enemy can melee
+        if (Controller.IsMelee)
+        {
+            // Checks the distance, whether a melee attack is happening and if line of sight isn't broken
+            if (Controller.Rays.PlayerSightBlocked == false)
+            {
+                // Checks if the enemy can do melee charges and whether it's in good range 
```

Comment:
This seems wordy and hard to read? Is there a better way this can be written for clarity?

*NFA at this point*